Greatly improve the System Requirements section of the docs (#593)
pulp/pulp_installer: docs/index.md

Before you install Pulp, review the [architecture and component documentation](https://docs.pulpproject.org/pulpcore/components.html#) to ensure you understand the deployment structure and concepts.

The Ansible [control node](https://docs.ansible.com/ansible/2.5/network/getting_started/basic_concepts.html#control-node)
(i.e., your workstation) must have Python 3 and Ansible (>= 2.9) installed.

Ensure that your server meets the [hardware requirements](https://docs.pulpproject.org/pulpcore/components.html#hardware-requirements) to install and run Pulp.

Ensure that your server, AKA the Ansible [managed node](https://docs.ansible.com/ansible/2.5/network/getting_started/basic_concepts.html#managed-nodes)
runs one of these currently supported operating systems:

- CentOS 7 or 8
- Debian Buster (needs `allow_world_readable_tmpfiles = True` in ansible.cfg)
- Fedora 32 or later

The server cannot provide any other HTTP (port 80, 443) service on the same hostname as Pulp's API. The only
exception is Pulp 2. The REST APIs for Pulp 2 and Pulp 3 can be served on the same hostname as
long as the `apache` webserver is deployed for both.

> These server requirements assume your are deploying Pulp to a single server. If you are deploying it

Reviewer: ```suggestion
These server requirements assume you are deploying Pulp to a single server. If you are deploying it
```

Author: Fixed.

to a cluster (with multiple tiers), the hardware requirements will differ, and only the webserver

> (pulp_webserver role) will have the service limitation. The OS of each node must be in the list.

Reviewer: Can we be specific about which list?

Author: Done.

The Ansible [control node](https://docs.ansible.com/ansible/2.5/network/getting_started/basic_concepts.html#control-node)
(i.e., your workstation) must have Python 3 and Ansible (>= 2.9) installed.

Ensure that your server meets the [hardware requirements](https://docs.pulpproject.org/pulpcore/components.html#hardware-requirements) to install and run Pulp.

Ensure that your server, AKA the Ansible [managed node](https://docs.ansible.com/ansible/2.5/network/getting_started/basic_concepts.html#managed-nodes),
runs one of these currently supported operating systems:

- CentOS 7 or 8
- Debian Buster (needs `allow_world_readable_tmpfiles = True` in ansible.cfg)
- Fedora 32 or later

The server cannot provide any other HTTP (port 80, 443) service on the same hostname as Pulp's API. The only
exception is Pulp 2. The REST APIs for Pulp 2 and Pulp 3 can be served on the same hostname as
long as the `apache` webserver is deployed for both.

NOTE: These server requirements assume you are deploying Pulp to a single server. If you are deploying it
to a cluster (with multiple tiers), the hardware requirements will differ, and only the webserver

> (pulp_webserver role) will have the service limitation. The OS of each node must be in the OS list
> above, but each node can run a different OS.

Reviewer: ```suggestion
(pulp_webserver role) will have the service limitation. Each node must run a supported operating system from the list above, but each node can run a different OS.
```

Oh I see what you're saying now! If you'll forgive me, I'd suggest this change^

Author: Incorporated.

> The Ansible collection requires [geerlingguy.postgresql](https://galaxy.ansible.com/geerlingguy/postgresql) role,

Reviewer 2: Is this still the only dependency? I thought, with the 2.10 we need some `community.*` collections too.

Author: I think I tested this, and the collection install command installs the dependency collections, but not the dependency rolls.

Author: Correct, it does not:
```
[mdepaulo@mdepaulo .ansible]$ pwd
/home/mdepaulo/.ansible
[mdepaulo@mdepaulo .ansible]$ ansible-galaxy collection install pulp.pulp_installer
Process install dependency map
Starting collection install process
Installing 'pulp.pulp_installer:3.12.1' to '/home/mdepaulo/.ansible/collections/ansible_collections/pulp/pulp_installer'
Skipping 'ansible.posix' as it is already installed
[mdepaulo@mdepaulo .ansible]$ ls roles/
adriagalin.motd/            lexa-uw.letsencrypt/
gzm55.smart_ssh_pipelining/ thulium_drake.acme_ssl/
[mdepaulo@mdepaulo .ansible]$
```
